Traceback (most recent call last):
  File "/Users/tompkins/Library/Python/3.9/lib/python/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/Users/tompkins/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 1093, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/tompkins/Library/Python/3.9/lib/python/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/tompkins/Library/Python/3.9/lib/python/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/Users/tompkins/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 559, in async_execute
    await self.async_execute_cell(
  File "/Users/tompkins/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 854, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/Users/tompkins/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 756, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# let's try it out 
tsfile=make_ts_yearmean(files)

# check if the file is there:
annmean=glob.glob(odir+"*fldmean_yearmean.nc")
print(annmean)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
[0;32m/var/folders/sh/kt3hqg9x639_js73vspx7ch40000gn/T/ipykernel_33966/4141558076.py[0m in [0;36m<module>[0;34m[0m
[1;32m      1[0m [0;31m# let's try it out[0m[0;34m[0m[0;34m[0m[0m
[0;32m----> 2[0;31m [0mtsfile[0m[0;34m=[0m[0mmake_ts_yearmean[0m[0;34m([0m[0mfiles[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m      3[0m [0;34m[0m[0m
[1;32m      4[0m [0;31m# check if the file is there:[0m[0;34m[0m[0;34m[0m[0m
[1;32m      5[0m [0mannmean[0m[0;34m=[0m[0mglob[0m[0;34m.[0m[0mglob[0m[0;34m([0m[0modir[0m[0;34m+[0m[0;34m"*fldmean_yearmean.nc"[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;32m/var/folders/sh/kt3hqg9x639_js73vspx7ch40000gn/T/ipykernel_33966/3988311038.py[0m in [0;36mmake_ts_yearmean[0;34m(files)[0m
[1;32m      3[0m [0;34m[0m[0m
[1;32m      4[0m     [0;31m# set cdo to an incidence of Cdo(), you don't need to do this, but cdo.func is neater than Cdo().func[0m[0;34m[0m[0;34m[0m[0m
[0;32m----> 5[0;31m     [0mcdo[0m [0;34m=[0m [0mCdo[0m[0;34m([0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m      6[0m [0;34m[0m[0m
[1;32m      7[0m     [0;31m# this illustrates one advantage of cdo within python, the ability to auto handle tmp files...[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/Library/Python/3.9/lib/python/site-packages/cdo.py[0m in [0;36m__init__[0;34m(self, cdo, returnNoneOnError, forceOutput, env, debug, tempdir, logging, logFile, cmd, options)[0m
[1;32m    154[0m     [0mself[0m[0;34m.[0m[0m_options[0m [0;34m=[0m [0moptions[0m[0;34m[0m[0;34m[0m[0m
[1;32m    155[0m [0;34m[0m[0m
[0;32m--> 156[0;31m     [0mself[0m[0;34m.[0m[0moperators[0m         [0;34m=[0m [0mself[0m[0;34m.[0m[0m__getOperators[0m[0;34m([0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    157[0m     [0mself[0m[0;34m.[0m[0mnoOutputOperators[0m [0;34m=[0m [0;34m[[0m[0mop[0m [0;32mfor[0m [0mop[0m [0;32min[0m [0mself[0m[0;34m.[0m[0moperators[0m[0;34m.[0m[0mkeys[0m[0;34m([0m[0;34m)[0m [0;32mif[0m [0;36m0[0m [0;34m==[0m [0mself[0m[0;34m.[0m[0moperators[0m[0;34m[[0m[0mop[0m[0;34m][0m[0;34m][0m[0;34m[0m[0;34m[0m[0m
[1;32m    158[0m     [0mself[0m[0;34m.[0m[0mreturnNoneOnError[0m [0;34m=[0m [0mreturnNoneOnError[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/Library/Python/3.9/lib/python/site-packages/cdo.py[0m in [0;36m__getOperators[0;34m(self)[0m
[1;32m    234[0m     [0moperators[0m [0;34m=[0m [0;34m{[0m[0;34m}[0m[0;34m[0m[0;34m[0m[0m
[1;32m    235[0m [0;34m[0m[0m
[0;32m--> 236[0;31m     [0mversion[0m [0;34m=[0m [0mparse_version[0m[0;34m([0m[0mgetCdoVersion[0m[0;34m([0m[0mself[0m[0;34m.[0m[0mCDO[0m[0;34m)[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    237[0m     [0;32mif[0m [0;34m([0m[0mversion[0m [0;34m<[0m [0mparse_version[0m[0;34m([0m[0;34m'1.7.2'[0m[0;34m)[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m    238[0m       [0mproc[0m [0;34m=[0m [0msubprocess[0m[0;34m.[0m[0mPopen[0m[0;34m([0m[0;34m[[0m[0mself[0m[0;34m.[0m[0mCDO[0m[0;34m,[0m [0;34m'-h'[0m[0;34m][0m[0;34m,[0m [0mstderr[0m[0;34m=[0m[0msubprocess[0m[0;34m.[0m[0mPIPE[0m[0;34m,[0m [0mstdout[0m[0;34m=[0m[0msubprocess[0m[0;34m.[0m[0mPIPE[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/Library/Python/3.9/lib/python/site-packages/cdo.py[0m in [0;36mgetCdoVersion[0;34m(path2cdo, verbose)[0m
[1;32m     76[0m     [0;32mreturn[0m [0mcdo_help[0m[0;34m[0m[0;34m[0m[0m
[1;32m     77[0m   [0mmatch[0m [0;34m=[0m [0mre[0m[0;34m.[0m[0msearch[0m[0;34m([0m[0;34m"Climate Data Operators version (\d.*) .*"[0m[0;34m,[0m [0mcdo_help[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 78[0;31m   [0;32mreturn[0m [0mmatch[0m[0;34m.[0m[0mgroup[0m[0;34m([0m[0;36m1[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     79[0m [0;34m[0m[0m
[1;32m     80[0m [0;32mdef[0m [0msetupLogging[0m[0;34m([0m[0mlogFile[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m

[0;31mAttributeError[0m: 'NoneType' object has no attribute 'group'
AttributeError: 'NoneType' object has no attribute 'group'

